work_queue/src/work_queue_worker.c

```diff
@@ -78,6 +78,12 @@ typedef enum {
 	CONTAINER_MODE_UMBRELLA
 } container_mode_t;
 
+struct environment {
+	char *tarball;     /* Conda environemnt as produced by conda-pack. */
+	char *expansion;   /* Directory in cache/. Only set when environemnt has been expanded.*/
+	int error;         /* Whether the expansion had an error. 1 for yes, 0 for no. */
```

Comment:
Booleans always seem to grow beyond their original purpose.
Let's call this `status` and have an enumeration for the possible states.